spharpy/spherical.py

```diff
@@ -1115,3 +1118,385 @@ def sph_identity_matrix(n_max, type='n-nm'):
+        objects with sampling points for which the basis matrix is
+        calculated
+    basis_type : str, optional
+        Type of spherical harmonic basis, either ``'complex'`` or
+        ``'real'``. The default is ``'real'``.
+    normalization : str, optional
+        Normalization convention, either ``'n3d'``, ``'maxN'`` or
+        ``'sn3d'``. The default is ``'n3d'``.  
+        (maxN is only supported up to 3rd order)
+    channel_convention : str, optional
+        Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+        The default is ``'acn'``.
+        (FuMa is only supported up to 3rd order)
+    inverse_method : str, optional
+        Inverse transform type, either ``'pseudo_inverse'`` or
+        ``'quadrature'``. The default is "pseudo_inverse".
+    condon_shortley : bool or str, optional
+        Whether to include the Condon-Shortley phase term. If ``True``,
+        Condon-Shortley is included, if ``False`` it is not
+        included. The default is ``'auto'``, which corresponds to
+        ``True`` for complex basis and ``False`` for real basis.
+        
+    """
+
+    def __init__(
+        self,
+        n_max,
+        coordinates,
+        basis_type="real",
+        normalization="n3d",
+        channel_convention="acn",
+        inverse_method="auto",
+        condon_shortley='auto',
+    ):
+        # initialize private attributes
+        self._n_max = None
+        self._coordinates = pf.Coordinates()
+        self._basis_type = None
+        self._inverse_method = None
+        self._channel_convention = None
+        self._condon_shortley = None
+        self._normalization = None
+        self._reset_compute_attributes()
+
+        self.n_max = n_max
+        self.coordinates = coordinates
+        self.basis_type = basis_type
+        self.inverse_method = inverse_method
+        self.channel_convention = channel_convention
+        self.normalization = normalization
+        self.condon_shortley = condon_shortley
+
+    # Properties
+    @property
+    def condon_shortley(self):
+        """Get or set the Condon-Shortley phase term."""
+        return self._condon_shortley
+
+    @condon_shortley.setter
+    def condon_shortley(self, value):
+        """Get or set the Condon-Shortley phase term."""
+        if isinstance(value, str):
+            if value != 'auto':
+                raise ValueError("condon_shortley must be a bool or the string 'auto'")
+        elif not isinstance(value, bool):
+            raise TypeError("condon_shortley must be a bool or the string 'auto'")
+        if value != self._condon_shortley:
+            self._reset_compute_attributes()
+        self._condon_shortley = value
+
+    @property
+    def n_max(self):
+        """Get or set the spherical harmonic order."""
+        return self._n_max
+
+    @n_max.setter
+    def n_max(self, value):
+        """Set the spherical harmonic order."""
```

Comment:
I think you forgot to copy the docstring of the setter to the getter in this case